app/views/works/index.html.erb

```diff
@@ -0,0 +1,41 @@
+<% books = @works.where(category: "book") %>
+<% albums = @works.where(category: "album") %>
+<h2>List of Works</h2>
+<h3>Books</h3>
+<table>
+    <tr>
+        <th>Votes</th>
+        <th>Title</th>
+        <th>Created by</th>
+        <th>Published</th>
+        <th>Upvote</th>
+    </tr>
+    <% books.each do |book| %>
+        <tr>
+            <td><%= book.vote_count %></td>
+            <td><%= link_to book.title, work_path(book) %></td>
+            <td><%= book.creator %></td>
+            <td><%= book.publication_year %></td>
+            <td><%= link_to "Upvote", vote_path(book.id) %></td>
```

Comment:
This link produces the error `No route matches [GET] "/works/17/vote"`. The default method for `link_to` is get, so you need to specify the method.
```suggestion
            <td><%= link_to "Upvote", vote_path(book.id), method: :post %></td>
```

Reply:
With this change, the `upvote` button works!